tools/cli/commands/changelog.js

```diff
@@ -512,25 +518,31 @@ async function gitAdd( argv ) {
 }
 
 /**
- * Gets list of currently modified files.
+ * Checks if changelog files are required.
  *
- * @returns {Array} modifiedProjects - projects that need a changelog.
+ * @returns {Array} matchedProjects - projects that need a changelog.
  */
-async function changedProjects() {
-	const re = /^projects\/([^/]+\/[^/]+)\//; // regex matches project file path, ie 'project/packages/connection/..'
-	const modifiedProjects = new Set();
-	const git = simpleGit();
-	const gitStatus = await git.status();
-	for ( const file of gitStatus.files ) {
-		const match = file.path.match( re );
-		if ( match ) {
-			modifiedProjects.add( match[ 1 ] );
-		}
+async function checkChangelogFiles() {
+	console.log( chalk.green( 'Checking if changelog files are needed. Just a sec...' ) );
+
+	// Bail if we're pushing to a release branch, like boost/branch-1.3.0
+	let currentBranch = child_process.spawnSync( 'git', [ 'branch', '--show-current' ] );
+	currentBranch = currentBranch.stdout.toString().trim();
+	const branchReg = /.*\/branch-(\d+).(\d+)(.(\d+))?/; // match example: jetpack/branch-1.2.3
+	if ( currentBranch.match( branchReg ) ) {
+		console.log( chalk.green( 'Release branch detected. No changelog required.' ) );
+		return [];
 	}
 
-	return allProjects().filter( proj => modifiedProjects.has( proj ) );
-}
+	const needChangelog = child_process.spawnSync( 'tools/check-changelogger-use.php', [
```

Comment:
That's a good question and something I hadn't considered yet. I have to do some more testing, but I'd expect `jetpack changelog add` would still tell you to add a changelog since it's checking against `origin/master`. But we can probably change what we're checking against when running the script (which would basically tell us if the feature branch added a changelog file)

Reply:
Considering a case like `master → feature branch → some commit to be merged into the feature branch`: if "feature branch" is adding a change entry for a project, then "some commit to be merged into the feature branch" will _not_ be prompted to add another change entry since it's looking at the whole diff from master up to HEAD.

If we did want it to prompt, we'd need to somehow detect that "feature branch" exists in the middle there and use that instead of origin/master as the base ref. Which would probably be error prone to do automatically.

Reply:
> will not be prompted to add another change entry since it's looking at the whole diff from master up to HEAD.

This sounds like expected behavior to me, so I think we're okay to leave it as is, especially if we implement other suggestions that let you add it pre-commit as well.